secret-script.py

```diff
@@ -1,64 +1,82 @@
-flags.DEFINE_boolean('g2k', False, 'Secret sync from Kubernetes to Gcloud')
-flags.DEFINE_boolean('create', None, 'Create new secret with secret_id')
-flags.DEFINE_boolean('delete', None, 'Delete secret with secret_id')
-flags.DEFINE_boolean('get', None, 'Get secret with secret_id')
-flags.DEFINE_string('file', None, 'Create or update from file')
-flags.DEFINE_string('secret_id', None, 'The string id of the secret')
-
-flags.mark_flag_as_required("secret_id")
 
 from gcloud_utils import *
 from k8s_utils import *
+import argparse
+import sys
+
+def parse_args(args):
+	parser = argparse.ArgumentParser()
+	parser.add_argument('action', help='Options: create, get, update, or delete')
+	parser.add_argument('--secret_id', help='The id of the secret')
+	parser.add_argument('--file', help='The yaml file containing the secret')
+	parser.add_argument('--direction', help='Options: k2g or g2k')
+	return parser.parse_args(args)
+
 
+def main(args):
+	# create secrets <args.secret_id> with file <args.file>
+	if args.action == 'create':
+		if args.secret_id is None or args.file is None:
+			sys.exit("requires '--secret_id' and '--file' arguments")
+		gcloudCreateSecret(args.secret_id)
+		gcloudAddSecrVersion(args.secret_id, args.file)
+		k8sCreateSecret(args.secret_id, args.file)
 
-def main(argv):
-	if len(argv) > 1:
-		raise app.UsageError('Too many command-line arguments.')
+	# get secrets <args.secret_id>
+	elif args.action == 'get':
+		if args.secret_id is None:
+			sys.exit("requires '--secret_id' arguments")
+		gcloudAccessSecretVersion(args.secret_id, "latest")
+		k8sAccessSecret(args.secret_id)
+		print("=============================")
 
-	if FLAGS.create is not None:
-		gcloudCreateSecret(FLAGS.secret_id)
-		gcloudAddSecrVersion(FLAGS.secret_id, FLAGS.file)
-		k8sCreateSecret(FLAGS.secret_id, FLAGS.file)
 
-	elif FLAGS.get is not None:
-		gcloudAccessSecretVersion(FLAGS.secret_id, "latest")
-		k8sAccessSecret(FLAGS.secret_id)
+	# delte secrets <args.secret_id>
+	elif args.action == 'delete':
+		if args.secret_id is None :
+			sys.exit("requires '--secret_id' argument")
+		gcloudDeleteSecret(args.secret_id)
+		k8sDeleteSecret(args.secret_id)
 
+	# update secrets <args.secret_id> with file <args.file> in a platform and sync to the other platform
+	# k2g: update kubernetes secret first, then sync to gcloud sm
+	# g2k: update gcloud sm secret first, then sync to kubernetes
+	elif args.action == 'update':
+		if args.secret_id is None or args.file is None:
+			sys.exit("requires '--secret_id' and '--file' arguments")
 
-	elif FLAGS.delete is not None:
-		gcloudDeleteSecret(FLAGS.secret_id)
-		k8sDeleteSecret(FLAGS.secret_id)
+		if args.direction == "k2g":
+			print("Update k8s secret: ")
+			k8sUpdateSecret(args.secret_id, args.file)
+			new_secret = k8sAccessSecret(args.secret_id)
+			gcloudAccessSecretVersion(args.secret_id, "latest")
+			
+			# sync
+			print("\n Synchronizing...\n")
```

Comment:
It would be useful for demo purposes to describe what you're syncing from and to

Reply:
Add from and to in the synchronizing message.